File: haskell/equalwords.hs
import Data.List
import GHC.Exts

main = do
    input <- readFile "../words.txt"
    let words = lines input
        anagrams = groupWith (sort) words
        formattedLines = map (\words -> intercalate ", " words) anagrams

    mapM (putStrLn) formattedLines

File: go/equalwords.go
package main

import(
    "fmt"
    "io/ioutil"
    "strings"
    "sort"
)

func main() {
    content, _  := ioutil.ReadFile("words.txt")
    var anagrams = make(map[string][]string)
    lines := strings.Split(string(content), "\n")

    for _,line := range lines{
        if(line == ""){
            break;
        }

        var sortedLine = strings.Split(line, "")
        sort.Strings(sortedLine);
        var sortedLineAsString = strings.Join(sortedLine, "")
        anagrams[sortedLineAsString] = append(anagrams[sortedLineAsString], line)
    }

    for _, v := range anagrams{
        fmt.Println(strings.Join(v, ", "))
    }
}
